lib/ocpp/v201/database_handler.cpp

```diff
@@ -743,18 +759,197 @@ void DatabaseHandler::insert_or_update_charging_profile(const int evse_id, const
     }
 }
 
-void DatabaseHandler::delete_charging_profile(const int profile_id) {
+bool DatabaseHandler::delete_charging_profile(const int profile_id) {
     std::string sql = "DELETE FROM CHARGING_PROFILES WHERE ID = @profile_id;";
     auto stmt = this->database->new_statement(sql);
 
     stmt->bind_int("@profile_id", profile_id);
     if (stmt->step() != SQLITE_DONE) {
         throw QueryExecutionException(this->database->get_error_message());
     }
+
+    return stmt->changes() > 0;
+}
+
+bool DatabaseHandler::clear_charging_profiles() {
+    return this->database->clear_table("CHARGING_PROFILES");
+}
+
+bool DatabaseHandler::clear_charging_profiles_matching_criteria(const std::optional<int32_t> profile_id,
+                                                                const std::optional<ClearChargingProfile>& criteria) {
+    // K10.FR.03, K10.FR.09
+    if (profile_id.has_value()) {
+        return this->delete_charging_profile(profile_id.value());
+    }
+
+    // criteria has no value, so clear all
+    if (!criteria.has_value()) {
+        return this->clear_charging_profiles();
```

Comment:
This flow is giving me serious pause. I don't like the absence of a specific intent being equated to the intent to destroy everything. Can you talk me through this flow? This seems like a serious side effect.